src/System.Runtime.Extensions/tests/System/Math.cs

```diff
@@ -1,134 +1,654 @@
+            }
+            else if (double.IsNaN(actual))
+            {
+                throw new EqualException(ToString(expected), ToString(actual));
+            }
+
+            if (double.IsNegativeInfinity(expected))
+            {
+                if (double.IsNegativeInfinity(actual))
+                {
+                    return;
+                }
+
+                throw new EqualException(ToString(expected), ToString(actual));
+            }
+            else if (double.IsNegativeInfinity(actual))
+            {
+                throw new EqualException(ToString(expected), ToString(actual));
+            }
+
+            if (double.IsPositiveInfinity(expected))
+            {
+                if (double.IsPositiveInfinity(actual))
+                {
+                    return;
+                }
+
+                throw new EqualException(ToString(expected), ToString(actual));
+            }
+            else if (double.IsPositiveInfinity(actual))
+            {
+                throw new EqualException(ToString(expected), ToString(actual));
+            }
+
+            if (IsNegativeZero(expected))
+            {
+                if (IsNegativeZero(actual))
+                {
+                    return;
+                }
+
+                if (IsPositiveZero(variance) || IsNegativeZero(variance))
+                {
+                    throw new EqualException(ToString(expected), ToString(actual));
+                }
+
+                // When the variance is not ±0.0, then we are handling a case where
+                // the actual result is expected to not be exactly -0.0 on some platforms
+                // and we should fallback to checking if it is within the allowed variance instead.
+            }
+            else if (IsNegativeZero(actual))
+            {
+                // Do nothing, since the actual result could still be within the allowed variance
+            }
+
+            if (IsPositiveZero(expected))
+            {
+                if (IsPositiveZero(actual))
+                {
+                    return;
+                }
+
+                if (IsPositiveZero(variance))
+                {
+                    throw new EqualException(ToString(expected), ToString(actual));
+                }
+
+                // When the variance is not ±0.0, then we are handling a case where
+                // the actual result is expected to not be exactly +0.0 on some platforms
+                // and we should fallback to checking if it is within the allowed variance instead.
+            }
+            else if (IsPositiveZero(actual))
+            {
+                // Do nothing, since the actual result could still be within the allowed variance
+            }
+
+            var delta = Math.Abs(actual - expected);
+
+            if (delta > variance)
+            {
+                throw new EqualException(ToString(expected), ToString(actual));
```

Comment:
I didn't include it here because the xUnit EqualException doesn't have a way to easily expose the information.

However, we are essentially checking for a perfect match. That is, double can only display 15-17 digits, and we require the result to be within accurate to within 50/52 explicit mantissa bits.